VocaDb.ResXFileCodeGenerator.Tests/GithubIssues/Issue3/GeneratorTests.cs

```diff
@@ -116,4 +117,141 @@ public static class CommonMessages
 		);
 		source.ReplaceLineEndings().Should().Be(expected.ReplaceLineEndings());
 	}
+
+	[Fact]
+	public void Generate_StringBuilder_Name_PartialXmlWorks()
+	{
+		var text = $@"<?xml version=""1.0"" encoding=""utf-8""?>
+<root>
+  <data name=""Works"" xml:space=""preserve"">
+    <value>Works.</value>
+  </data>
+</root>";
+
+		var expected = $@"// ------------------------------------------------------------------------------
+// <auto-generated>
+//     This code was generated by a tool.
+//
+//     Changes to this file may cause incorrect behavior and will be lost if
+//     the code is regenerated.
+// </auto-generated>
+// ------------------------------------------------------------------------------
+#nullable enable
+namespace VocaDb.Web.App_GlobalResources
+{{
+    using System.Globalization;
+    using System.Resources;
+
+    public static class CommonMessages
+    {{
+        private static ResourceManager? s_resourceManager;
+        public static ResourceManager ResourceManager => s_resourceManager ??= new ResourceManager(""VocaDb.Web.App_GlobalResources.CommonMessages"", typeof(CommonMessages).Assembly);
+        public static CultureInfo? CultureInfo {{ get; set; }}
+
+        /// <summary>
+        /// Looks up a localized string similar to Works..
+        /// </summary>
+        public static string? Works => ResourceManager.GetString(nameof(Works), CultureInfo);
+    }}
+}}";
+		var generator = new StringBuilderGenerator();
+		using var resxStream = new StringReader(text);
+		var source = generator.Generate(
+			resxStream: resxStream,
+			options: new GeneratorOptions(
+				LocalNamespace: "VocaDb.Web.App_GlobalResources",
+				CustomToolNamespace: null,
+				ClassName: "CommonMessages",
+				PublicClass: true,
+				NullForgivingOperators: false,
+				StaticClass: true
+			)
+		);
+		source.ReplaceLineEndings().Should().Be(expected.ReplaceLineEndings());
+	}
+
+	[Fact]
+	public void Generate_StringBuilder_Name_DuplicatedataGivesWarning()
+	{
+		var text = $@"<?xml version=""1.0"" encoding=""utf-8""?>
+<root>
+  <data name=""DupKey"" xml:space=""preserve"">
+    <value>Works.</value>
+  </data>
+  <data name=""DupKey"" xml:space=""preserve"">
+    <value>Doeesnt Work.</value>
+  </data>
+</root>";
+
+		var generator = new StringBuilderGenerator();
+		using var resxStream = new StringReader(text);
+		List<Diagnostic> errs = new();
+
+		var source = generator.Generate(
+			resxStream: resxStream,
+			options: new GeneratorOptions(
+				LocalNamespace: "VocaDb.Web.App_GlobalResources",
+				CustomToolNamespace: null,
+				ClassName: "CommonMessages",
+				PublicClass: true,
+				NullForgivingOperators: false,
+				StaticClass: true,
+				ReportError: e=>errs.Add(e),
```

Comment:
Style nit: Please place spaces before and after the `=>` operator.

Reply:
done